xbmc/music/MusicDatabase.cpp

```diff
@@ -3600,8 +3600,8 @@ bool CMusicDatabase::GetRolesNav(const std::string& strBaseDir, CFileItemList& i
       return false;
 
     // get roles with artists having that role
-    std::string strSQL = "SELECT idRole, strRole FROM role WHERE EXISTS "
-                         "(SELECT 1 FROM song_artist WHERE song_artist.idRole = role.idRole)";
+    std::string strSQL = "SELECT DISTINCT role.idRole, role.strRole FROM role "
+                         "JOIN song_artist ON song_artist.idRole = role.idRole ";
```

Comment:
I tested on my i7 (with fake data) and I got a better result in both Mysql and sqlite.. But I guess it's a quick query anyway so shouldn't be a big problem even if little bit slower..